### DISCUSSION TITLE
dynamically display the mkdocs-material runnig host IP in generated HTML

### DISCUSSION THREAD

**Discussion by obuno** [squidfunk/mkdocs-material#7298] [Q&A]
Hi team,

I'm currently working on some documentation and would like whenever possible to "automate" the inclusion/display of the IP address of the server running the mkdocs-material generated site.

Let me illustrate this with a .md snippet code example:

```
  !!! example "Interacting with the STUDENT host"
  
      You can obtain a bash shell toward the STUDENT host from an SSH session.    
      I.E. launching an SSH session from a Linux based endpoint: ```ssh -l root -p 2222 YOUR-IP/FQDN```
```

In the above code, if ever possible, I'd like to replace the `YOUR-IP/FQDN` with the actual public IP address of the host serving the documentation. Obviously, that information can change/is volatile as multiple instances are made available at the same time, one per students that is..

Anyone would have ideas on how to solve this? 

Thanks a lot for MKdocs,
Cheers

**Comment by alexvoss** [comment]
Do I understand you correctly that you are compiling different versions of the site, N times for N students? That would mean you know the IP address at compile time? If that is the case then perhaps the [mkdocs-macros plugin](https://mkdocs-macros-plugin.readthedocs.io/en/stable/) would do the trick? 

Given that the students are taught to use `ssh`, should they not also be able to figure out the IP address or FQDN themselves? I have written instructions like this and never felt the need to do that part for the students. Now, perhaps you are doing this for the sake of efficiency if there is little time.

**Reply by alexvoss** [reply]
Something like this:

**dependencies**:

```
pip install mkdocs-macros-plugin
```

**mkdocs.yml**:

```
...
plugins:
    - search
    - macros
```

**main.py** (next to the `mkdocs.yml`)

```python
import socket

def define_env(env):
    """ 
    This is the hook for the functions (new form)
    """

    @env.macro
    def get_fqdn():
        return socket.getfqdn()
```

**Reply by obuno** [reply]
Hi Alex, 

thanks a lot for your update -- much appreciated. Yes you got that right, same env/same docs + unique public IP per student.

Indeed, as you mentioned this is a bit over babysitting. Clearly they (the students) will be able to figure out their own POD's IP's/FQDN's for the vast majority of course. This was more for me to egg kill off my desk questions like "What is the IP of my POD" =)

Thanks for the ```mkdocs-macro-plugin``` tip, I'll test this out.

Cheers
